### PR TITLE
doc: CF-1565 Documentation for new api version

### PR DESCRIPTION
**Pull request by aym7**
**Description**
CF-1565
I splitted the doc in several readme and folders. To try and have a versionned documentation. 

If possible try and read it from the PoV of someone external to our codebase, as it's directed to clients

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,16 +143,39 @@ Examples:
...
+GET /<data_type>
+```
+### Response
+The response is an array of all data instances of `data_type` owned by the user's company.
+
+[//]: # (TODO: Update payloads)
+See [product_list_sample.json](versionned/v1/v1-product_list_sample.json) for an example on what could be returned.
+> The child data are not all returned, to reduce the amount of information returned.
+
+## Get a specific element
```
Comment (NaughtyGnosiophile) [inline README.md:156]:
```suggestion
## Get a specific resource
```
These are commonly known as "resources" in the REST world. "Element" often refers to specific fields in the payload, which would be confusing.

File: README.md [hunk 2]
```diff
@@ -126,16 +143,39 @@ Examples:
...
+
 A valid json with the data fields.
 
-You can see an example for products in [product_payload_sample.json](product_payload_sample.json).  
-See [product_payload_definition.json](product_payload_definition.json) for the payload format.
+You can see an example for products in [product_payload_sample.json](versionned/v1/v1-product_payload_sample.json).  
+See [product_payload_definition.json](versionned/v1/v1-product_payload_definition.json) for the payload format.
 
 - `base64_image_file`: replace by a base64 encoded image file, smaller than 5MB ("jpeg", "jpg" or "png").
 - `base64_pdf_file`: replace by a base64 encoded file, smaller than 5MB ("pdf", "jpeg", "jpg or "png").
```
Comment (NaughtyGnosiophile) [inline README.md:178]: In the payload json examples, we might want to put `<base64_image_file>` to make it clear this is a "stand in" marker.

**Review by NaughtyGnosiophile — CHANGES_REQUESTED** [review]
We'll also need to fill in the v1 sample payloads. Should be easy by just doing at GET against a `db_seed` object.

File: README.md [hunk 3]
```diff
@@ -126,16 +143,39 @@ Examples:
 * `/products/4` to get the product with id **4**.
 * `/products/4/variations/3` for the variation with id **3**, in the product with id **4**
 
+## List
+Generic endpoint, `GET` method.
```
[the remark this answers is not in the document]
Comment (aym7) [reply README.md:146]: I've reworked the examples and reorganised them. I also used the List/Detail terminology :)

**Review by NaughtyGnosiophile — APPROVED** [review]
🚢

**Pull request merged**